Library/Homebrew/test/language/node_spec.rb

```diff
@@ -0,0 +1,36 @@
+require "language/node"
+
+describe Language::Node do
+  describe "#setup_npm_environment" do
+    it "does nothing when npmrc exists" do
+      expect(subject.setup_npm_environment).to be_nil
+    end
+
+    it "calls prepend_path when npmrc does not exist" do
+      allow(Formula).to receive(:[]).with("node").and_return(formula { url "foo-1.0" })
+      allow_any_instance_of(Pathname).to receive(:exist?).and_return(false)
+      allow(ENV).to receive(:prepend_path).with("PATH", Formula["node"].opt_libexec/"bin").and_return("prepend_path called")
```

Comment:
I think it would be good to avoid stubbing this method and just verify that the `ENV["PATH"]` includes the expected `node` path. The less stubbing we can do, the better. We want to use stubs to control only for variation in the users who run the tests.